common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -125,10 +125,13 @@ public static void load(String originalName, ClassLoader loader) {
             }
         }
 
-        if (url == null) {
-            // Fall back to normal loading of JNI stuff
+        try {
+            // first try to load from java.library.path
             loadLibrary(loader, name, false);
+            logger.info("{} was loaded from java.libary.path", name);
             return;
+        } catch (Exception ex) {
+            logger.debug("{} cannot be loaded from java.libary.path", name, ex);
```

Comment:
Regarding to the order: I think in terms of being cautious with resource usage, trying to load an existing file first before we write something to /tmp is the better approach. On my system, writing to /tmp will work but loading the lib will fail, resulting in an unneeded tmp file write.